### ISSUE TITLE
SqlServer: A lot of processing job - timeout

### ISSUE THREAD

**Issue by knopa** [HangfireIO/Hangfire#1733]
I got for one day 4kk processing job. When trying to check it via dashboard always got
The wait operation timed out

`[SqlException (0x80131904): Execution Timeout Expired.  The timeout period elapsed prior to completion of the operation or the server is not responding.]
   System.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction) +54995377
   System.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose) +764
   System.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady) +4243
   System.Data.SqlClient.SqlDataReader.TryConsumeMetaData() +90
   System.Data.SqlClient.SqlDataReader.get_MetaData() +103
   System.Data.SqlClient.SqlCommand.FinishExecuteReader(SqlDataReader ds, RunBehavior runBehavior, String resetOptionsString, Boolean isInternal, Boolean forDescribeParameterEncryption, Boolean shouldCacheForAlwaysEncrypted) +621
   System.Data.SqlClient.SqlCommand.RunExecuteReaderTds(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, Boolean async, Int32 timeout, Task& task, Boolean asyncWrite, Boolean inRetry, SqlDataReader ds, Boolean describeParameterEncryptionRequest) +3386
   System.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, String method, TaskCompletionSource`1 completion, Int32 timeout, Task& task, Boolean& usedCache, Boolean asyncWrite, Boolean inRetry) +710
   System.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, String method) +83
   System.Data.SqlClient.SqlCommand.ExecuteScalar() +2218
   Dapper.SqlMapper.ExecuteScalarImpl(IDbConnection cnn, CommandDefinition& command) +301
   Dapper.SqlMapper.ExecuteScalar(IDbConnection cnn, String sql, Object param, IDbTransaction transaction, Nullable`1 commandTimeout, Nullable`1 commandType) +123
   Hangfire.SqlServer.SqlServerMonitoringApi.GetNumberOfJobsByStateName(DbConnection connection, String stateName) +259
   Hangfire.SqlServer.SqlServerStorage.UseConnection(DbConnection dedicatedConnection, Func`2 func) +73
   Hangfire.SqlServer.SqlServerMonitoringApi.SucceededListCount() +74
   Hangfire.Dashboard.Pages.SucceededJobs.Execute() +361
   Hangfire.Dashboard.RazorPage.TransformText(String body) +31
   Hangfire.Dashboard.RazorPageDispatcher.Dispatch(DashboardContext context) +89
   Hangfire.Dashboard.<>c__DisplayClass1_1.<UseHangfireDashboard>b__1(IDictionary`2 env) +603
   Microsoft.Owin.Mapping.<Invoke>d__0.MoveNext() +462`

**Comment by jcspader** [comment]
I have load of 25k jobs daily.

Our problem can be your settings and/or version.
Give us more details on that. What page you receive timeout?

**Comment by mellamokb** [comment]
Agreed we have 50-100k jobs on our busiest days.  Currently sitting at 20k scheduled and 7M successful (1 month sliding window), dashboard is almost immediate responsive.  Though most of our jobs process at night so we don't see high actual processing volume during the day.

Hangfire is known to be pretty intensive on your database server, I would check that it is handling the load on your highest volume.  You might also try reducing the number of active workers, default is 5 per processor.  We ended up reducing that to 2 per processor (quad-core server so 8 workers) due to load/timeout issues at high volume.
